### PR TITLE
add channel and add subscribers 

### PR DESCRIPTION
**Pull request by MaiUbeid**
add channel and add subscribers with its query, controller and testing the router and query
get Channels in api/v1/getChannels

#40

### CODE REVIEW

File: server/controllers/messages/addChannel.js [hunk 1]
```diff
@@ -1,3 +1,10 @@
-exports.addChannel = (req, res) => {
-  res.send('Hi from posting channels');
+const { messages: { addChannel, addSubscriber } } = require('../../database/queries');
+
+exports.addChannel = (req, res, next) => {
+  const { channelName, subscribers } = req.body;
+  addChannel(channelName)
+    .then(({ rows: [{ id: channelId }] }) => addSubscriber(channelId, subscribers))
+    .then(() => res.json({ isSuccess: true, message: 'Channel is created with id:x' }))
```
Comment (Fatmasiam) [inline server/controllers/messages/addChannel.js]:
```suggestion
    .then(() => res.json({ isSuccess: true, message: 'Channel is created ' }))
```

File: test/queries/languages.test.js [hunk 2]
```diff
@@ -13,9 +13,8 @@ test('test Languages query', (t) => {
   ];
   dbBuild()
     .then(() => getLanguages())
-    .then((res) => res.rows)
-    .then((result) => {
-      t.deepEqual(result, expected, 'The table should be languages');
+    .then(({ rows }) => {
+      t.deepEqual(rows, expected, 'The table should be languages');
```
Comment (Fatmasiam) [inline test/queries/languages.test.js]: unrelated file

File: server/controllers/messages/addChannel.js [hunk 3]
```diff
@@ -1,3 +1,10 @@
-exports.addChannel = (req, res) => {
-  res.send('Hi from posting channels');
+const { messages: { addChannel, addSubscriber } } = require('../../database/queries');
+
+exports.addChannel = (req, res, next) => {
+  const { channelName, subscribers } = req.body;
+  addChannel(channelName)
```
Comment (Fatmasiam) [inline server/controllers/messages/addChannel.js]: where is the validation :open_mouth:

File: server/database/queries/messages/addChannel.js [hunk 4]
```diff
@@ -1,9 +1,23 @@
 const dbConnection = require('../../config/dbConnection');
 
-exports.addChannel = () => {
+exports.addChannel = (channelName = null) => {
```
Comment (Fatmasiam) [inline server/database/queries/messages/addChannel.js]: this does not make any sense here  

```suggestion
exports.addChannel = (channelName ) => {
```

File: server/controllers/messages/getChannels.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const { messages: { getChannels } } = require('../../database/queries');
+
+exports.getChannels = (req, res, next) => {
+  getChannels()
+    .then(({ rows }) => res.send({ data: rows }))
```
Comment (Fatmasiam) [inline server/controllers/messages/getChannels.js]: we agreed to use `res.json` i think 
```suggestion
    .then(({ rows }) => res.json({ isSuccess: true,  data: rows }))
```

File: server/database/queries/messages/addChannel.js (hunk 4 above)
Comment (Amoodaa) [reply server/database/queries/messages/addChannel.js]: we talked about this one, if we send to the query nothing: `null` it wont put a name for the channel

Comment (Fatmasiam) [reply server/database/queries/messages/addChannel.js]: i know btw try to test this query alone by sending to it a null value 
i will see it in the morning 
@Amoodaa

File: test/queries/languages.test.js (hunk 2 above)
Comment (MaiUbeid) [reply test/queries/languages.test.js]: Yes, but I find that this is better to edit it in the same pull request 
I will add it in its branch

File: server/controllers/messages/addChannel.js (hunk 3 above)
Comment (MaiUbeid) [reply server/controllers/messages/addChannel.js]: I did it ^_^
But I have a question 
Are we need to ensure that channel name is not already existing or will do that in the front end ??
@Fatmasiam

Comment (Fatmasiam) [reply server/controllers/messages/addChannel.js]: @MaiUbeid 
We don't need to ensure if it's exists 
The channel name can repeat or the same for two-channel or more

File: server/database/queries/messages/addChannel.js (hunk 4 above)
Comment (MaiUbeid) [reply server/database/queries/messages/addChannel.js]: What is the problem of making the default null? Is it already null?
@Fatmasiam

Comment (11janci) [reply server/database/queries/messages/addChannel.js]: default is undefined. If the channelName is optional, it is good to write a comment on the function explaining that that's by design and not a mistake, also you can name the parameter `channelNameOpt` to make it more clear

Comment (Amoodaa) [reply server/database/queries/messages/addChannel.js]: they say the `undefined` is the default, and you put it as null only when you want it, as a dev, to be `null`
